### PR TITLE
[DEVX-1772] Bump cypress to v10.1.0

### PR DESCRIPTION
**Pull request by tianfeng92**

### CODE REVIEW

**Comment by alexplischke** [comment]
With most of the pipeline failing, it's hard to do a proper evaluation. Would be great if you could pair it up with some example jobs or a custom saucectl version that works 🙏

**Comment by tianfeng92** [comment]
@alexplischke can you take a look at this PR please? 🙏

File: src/cypress-runner.js [hunk 1]
```diff
@@ -155,6 +153,7 @@ const getCypressOpts = function (runCfg, suiteName) {
     opts.config.videoUploadOnPasses = true;
   }
 
+  const cypressCfg = require(cypressCfgFile);
```
[the remark this answers is not in the document]
Comment (tianfeng92) [reply src/cypress-runner.js]: If it's typescript, it'd fail. But after looking into the cypressCfg I found it's not required here. So I just remove this and pass it as an args to cypress.

[the remark this answers is not in the document]
Comment (tianfeng92) [reply src/cypress-runner.js]: But still got an issue 🤔 checking

[the remark this answers is not in the document]
Comment (tianfeng92) [reply src/cypress-runner.js]: works

File: src/cypress-runner.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const {afterRunTestReport} = require('@saucelabs/cypress-plugin');
 const report = async (results = {}, statusCode, browserName, runCfg, suiteName, startTime, endTime, metrics) => {
   // Prepare the assets
   const runs = results.runs || [];
-  let specFiles = runs.map((run) => run.spec.name);
+  let specFiles = runs.map((run) => path.basename(run.spec.name));
```
[the remark this answers is not in the document]
Comment (tianfeng92) [reply src/cypress-runner.js:13]: Just make it consistent for generating junit.xml. `run.spec.name` contains the entire path for each spec. While the generated junit files are under `__assets__` and like `__assets__/actions.cy.js.xml`. Without this, the reporter will try to find junit xml file like `/home/seluser/__project__/__assets__/cypress/e2e/examples/actions.cy.js` and fail

**Review by alexplischke — APPROVED** [review]

**Pull request merged**